source/nodejs/adaptivecards-designer/src/data.ts

```diff
@@ -172,15 +172,17 @@ export class ArrayData extends DataType {
     generateSampleData(): any {
         let result = [];
 
-        for (let i = 1; i <= 3; i++) {
-            result.push(this.dataType.generateSampleData());
+        if (this.dataType) {
+            for (let i = 1; i <= 3; i++) {
```

Comment:
Nit Nit nit: Any specific reason why you prefer initializing to 1 vs 0 to fashion the loop construct? _havent checked but assuming you use this same pattern throughout the code base? If not any reason to diverge here?_ (
```suggestion
            for (let i = 0; i < 3; i++) {
```

Also just for my understanding as i start to get familiar with the code, where is the hardcoded 3 coming from again? What does it signify? Thanks again

Reply:
Because this is just a counter, and the index isn't used anywhere. I tend to count from 1 to 3 in real life, not from 0 to 2 :)